OpenRA.Test/OpenRA.Game/PngTest.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.IO;
+using System.Linq;
+using NUnit.Framework;
+using OpenRA.FileFormats;
+using OpenRA.Graphics;
+using OpenRA.Primitives;
+
+[TestFixture]
+public class PngTests
+{
+	[Test]
+	public void Save_ShouldProduceValidPngFile()
+	{
+		// Arrange
+		var colors = Enumerable.Range(0, 256).Select(i => Color.FromArgb(i, i, i)).ToArray();
+		var png = new Png(new byte[10 * 20], SpriteFrameType.Indexed8, 10, 20, colors);
+
+		// Act
+		byte[] result = png.Save();
+
+		// Assert
+		Assert.IsTrue(Png.Verify(new MemoryStream(result)));
+	}
+
+	[Test]
+	public void Save_Method_Should_Write_Indexed8_Palette_If_256_Colors_Or_Less()
+	{
+		// Arrange
+		var colors = Enumerable.Range(0, 256).Select(i => Color.FromArgb(i, i, i)).ToArray();
+		var png = new Png(new byte[10 * 20], SpriteFrameType.Indexed8, 10, 20, colors);
+
+		// Act
+		var result = png.Save();
+
+		// Assert
+		var colorTypeByte = result[25]; // Byte at index 25 contains color type information
+		Assert.That(colorTypeByte, Is.EqualTo(0x03)); // 0x03 represents Indexed8 with a palette
```

Comment:
Hm, maybe something like

```suggestion
		// Assert:
		// The byte at index 25 contains color type information
		// and 0x03 represents Indexed8 with a palette
		var colorTypeByte = result[25];
		Assert.That(colorTypeByte, Is.EqualTo(0x03));
```

here and below?